lib/kracken/controllers/authenticatable.rb

```diff
@@ -38,8 +38,11 @@ def authenticate_user
       def authenticate_user!
         check_token_expiry!
         unless user_signed_in?
-          if request.format == :json
+          case request.format
+          when :json
             render json: {error: '401 Unauthorized'}, status: :unauthorized
+          when :js
+            head :unauthorized
```

Comment:
Unfortunately this `case` format doesn't work as expected. The reason is `request.format` returns a [`Mime::Type`](http://api.rubyonrails.org/classes/Mime/Type.html). And while case equality is defined, `Mime::Type#===`, it will not be called in this statement; instead `Symbol#===` is called:

```ruby
# From sauron

type = Mime::Type.lookup("application/json")
# => #<Mime::Type:0x00007ffd761be830 @hash=-3662196928665928964,
#                                    @string="application/json",
#                                    @symbol=:json,
#                                    @synonyms=[
#                                      "text/x-json",
#                                      "application/jsonrequest",
#                                      "application/vnd.rn+json"
#                                    ]>

type == :json
# => true

type === :json
# => true

:json === type
# => false

case type
when :json
  :working
else
  :fail
end
# => :fail
```

It looks like we have two options:

- use `if` / `elsif` / `else`
- cast the type to a symbol `case request.format.to_sym`

Personally, I prefer the `case` version. However, I feel like that may be brittle to Rails changes. So I think the most flexible is going to be the `if` / `elseif` / `else` version.

Reply:
I think at this point we can safely also convert the initial check to a guard clause:

```ruby
return if user_signed_in?
# NOTE: Due to `case` semantics we cannot use it here as `Symbol#===`
#       does not match `Mime::Type`
if request.format == :json
  render json: {error: '401 Unauthorized'}, status: :unauthorized
elsif request.format == :js
  head :unauthorized
# ...
```

Reply:
Too much magic! I'll see if I can update the specs to return the proper format object type and then change the logic to `if`/`elsif`/`else` as you suggest.